### ISSUE TITLE
Evaluation index problem

### ISSUE THREAD

**Issue by Bin-ze** [hu64/FFAVOD#3]
First of all, thank you for your help, I have solved the previous problem.
However, new problems have emerged. I used centernet as my baseline to train on this data set, using coco indicators, but I found that the indicators change very strangely during training. Loss decreases with training, but the accuracy of the network on the test set gets worse with training, which is far from the results in your paper. I did not use the cneternet you used, but I am sure that there is no problem with my network, and it can get normal results on the coco data set.
I used the ignore.txt file of the dataset to process the data, because I think that unlabeled instances may cause instability in network training. I don't know what kind of treatment you have done?
I currently have no way to get a relatively normal baseline indicator. It makes it difficult for me to continue my work. Do you have any good suggestions?
For small goals, have you made improvements to the official code (it does not seem to be from the paper)?
What index did you use during training, VOC or COCO?

_Originally posted by @Bin-ze in https://github.com/hu64/FFAVOD/issues/2#issuecomment-985287384_

**Comment by hu64** [comment]
I am not sure, but I suggest that you first make sure that your GT is correct. With incorrect GT, you might be training for the wrong objectives. Otherwise, I didn't do any special treatment that is not already in CenterNet, normalize your data, data augmentation, etc. For my part, I did not consider the ignore regions during training, only during evaluation. Although you might get better result if you do, indeed. It's a good test to do. For the CenterNet baseline in the paper, I did not do any improvements, no. And to answer your last question, I use the COCO metric to validate my results.

**Comment by Bin-ze** [comment]
I want to know which of the coco indicators is similar to this data set.
Because I got pretty good results on coco, but I can only get about 33% AP using matlab testing tools. And my AP on the coco data set exceeded the result obtained by matlab evaluation。

I also want to know that the baseline you compared is actually not object as points? It is another paper Centernet: Keypoint triplets for object detection?
But your spotNet improves on object as points?
Am i right? Means that object as points can't get the results of the baseline in your paper?

**Comment by hu64** [comment]
First your first question, I don’t know if we can make any correspondance between the coco evaluation and the matlab tool. Particularly because of the ignore regions. For your second question, my baseline is definitely objects as points! Sorry if that was confusing in the paper. I never used keypoint triplets myself.

**Comment by Bin-ze** [comment]
During your training, the training set did not consider the neglected area. During the test, the neglected area was considered, and the neglected area was removed from the picture.

I have never used the test set data of the 20 folders officially given during the training.

When using centernet as a baseline, the data set only runs out a result on this network without changing the hyper parameters of the network.

You use Hourglass 104network.

Am I right?

The coco indicator I asked means how much map accuracy on coco is equivalent to 51ap on matlab.？

发自我的iPhone

> 在 2021年12月4日，下午11:15，Hughes Perreault ***@***.***> 写道？：
> 
> First your first question, I don’t know if we can make any correspondance between the coco evaluation and the matlab tool. Particularly because of the ignore regions. For your second question, my baseline is definitely objects as points! Sorry if that was confusing in the paper. I never used keypoint triplets myself

**Comment by hu64** [comment]
Yes exactly you are right. And the reason I am saying there is no COCO to Matlab equivalent is that there will be various numbers of false positives in the ignore regions, which will then be removed by the Matlab tool. So it varies from time to time. You probably could code an evaluation tool similar to the Matlab tool in the code if you wanted to, it would then be more integrated.

**Comment by Bin-ze** [comment]
Do you mean that I don't need to process the test set, and Natlab will eliminate the predictions I generate at the neglect area of the test set?

My previous processing was to replace the neglected area in the test set picture directly with a 0 value through ignore.txt. Did this cause my detector performance to decline?

Do you mean that I don't need to do any processing of dataset pictures, and the test tool will complete a series of processing during the test to get the correct results?

发自我的iPhone

> 在 2021年12月5日，下午10:43，Hughes Perreault ***@***.***> 写道：
> 
> Yes exactly you are right. And the reason I am saying there is no COCO to Matlab equivalent is that there will be various numbers of false positives in the ignore regions, which will then be removed by the Matlab tool. So it varies from time to time. You probably could code an evaluation tool similar to the Matlab tool in the code if you wanted to, it would then be more integrated

**Comment by hu64** [comment]
I am not sure at all, you can test it, but my intuition is that inserting zeros in the image might decline your performance, indeed. The way I did it was doing nothing, and letting the Matlab tool get the correct result for me.
